src/components/EditSections/EditUserInfo/components/ProfilePicture/ProfilePicture.js

```diff
@@ -12,33 +12,47 @@ import {
 import { useStripes } from '@folio/stripes/core';
 
 import { useProfilePicture } from '../../../../../hooks';
-import { isAValidUUID } from '../../../../util/util';
+import { isAValidURL } from '../../../../util/util';
 import profilePicThumbnail from '../../../../../../icons/profilePicThumbnail.png';
 import css from '../../EditUserInfo.css';
+import ExternalLinkModal from '../ExternalLinkModal';
 
-const ProfilePicture = ({ label, profilePictureLink }) => {
+const ProfilePicture = ({ profilePictureId, form }) => {
+  const [profilePictureLink, setProfilePictureLink] = useState(profilePictureId);
+  const [externalLinkModalOpen, setExternalLinkModalOpen] = useState(false);
   const intl = useIntl();
   const stripes = useStripes();
-  const isProfilePictureLinkAURL = !isAValidUUID(profilePictureLink);
   const hasProfilePicture = Boolean(profilePictureLink);
-  const { isFetching, profilePictureData } = useProfilePicture({ profilePictureId: profilePictureLink });
+  const isProfilePictureLinkAURL = hasProfilePicture && isAValidURL(profilePictureLink);
   const hasAllProfilePicturePerms = stripes.hasPerm('ui-users.profile-pictures.all');
 
+  const { isFetching, profilePictureData } = useProfilePicture({ profilePictureId });
+
   const renderProfilePic = () => {
     const profilePictureSrc = isProfilePictureLinkAURL ? profilePictureLink : 'data:;base64,' + profilePictureData;
     const imgSrc = isFetching || !hasProfilePicture ? profilePicThumbnail : profilePictureSrc;
 
     return (
       <img
         data-testid="profile-picture"
-        id="profile-picture"
         className={css.profilePlaceholder}
         alt={intl.formatMessage({ id: 'ui-users.information.profilePicture' })}
         src={imgSrc}
       />
     );
   };
 
+  const toggleExternalLinkModal = () => {
+    setExternalLinkModalOpen(prev => !prev);
```

Comment:
Let's use more meaningful values (e.g. open, isOpen)

Reply:
As discussed, I am going with this notation, as it does the toggle job irrespective of the status of the modal - open or close.